### ISSUE TITLE
Rule proposal: `prefer-global-this`

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-unicorn#978]
[`globalThis`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis) is the unified way to access the "global" object.

This has the benefit of making code more cross-platform and more consistent.

`globalThis` requires Node.js 12, so we cannot add it to the recommended preset until April 2021 (date of Node.js 10 obsoletion).

## Fail

```js
window.foo; // Browser

self.foo; // Web Worker

global.foo; // Node.js
```

## Pass

```js
globalThis.foo;
```

**Comment by sindresorhus** [comment]
This is now accepted.

**Comment by fregante** [comment]
Can this auto fix to `this` when it's the same? "globalThis" is so awkward

**Comment by sindresorhus** [comment]
> Can this auto fix to this when it's the same? "globalThis" is so awkward.

In what situations are they the same?

> Also it should exclude window’s actual properties, like window.innerWidth

Yes, good catch. A list can be found here: http://developer.mozilla.org/en-US/docs/Web/API/Window For those, it should enforce `window` over `globalThis`.

**Comment by fregante** [comment]
> In what situations are they the same?

Not entirely sure, but I assume at the top level they always are? 
<img width="255" alt="Screen Shot" src="https://user-images.githubusercontent.com/1402241/104839098-aa3d3f00-5884-11eb-9e01-765c2d4f8666.png">

**Comment by sindresorhus** [comment]
@fregante The REPL doesn't represent a proper Node.js environment. Try running the same in a file instead.

**Comment by fregante** [comment]
Changed my mind. `globalThis` is more clear 👍

**Comment by fregante** [comment]
Also:

```js
Object.assign(window, {something: true});

Object.defineProperty(window, 'something', {
  value: true,
  writable: true,
});

Object.defineProperties(window, {
  something: {
    value: true,
    writable: true,
  },
});
```
